sentry_sdk/attachments.py

```diff
@@ -0,0 +1,51 @@
+import os
+import mimetypes
+
+from sentry_sdk._types import MYPY
+from sentry_sdk.envelope import Item
+
+if MYPY:
+    from typing import Optional
+
+
+class Attachment(object):
+    def __init__(
+        self,
+        bytes=None,  # type: Optional[bytes]
+        filename=None,  # type: Optional[str]
+        path=None,  # type: Optional[str]
+        content_type=None,  # type: Optional[str]
+        add_to_transactions=False,  # type: bool
+    ):
```

Comment:
I think I would opt to make two `from_path()` and `from_bytes` classmethods and encourage the users to use them rather than play the which-args-go-together python guessing game

Reply:
Generally I would not expose the use of this class yet. Leaving rest of comments below.